### PR TITLE
Fix warnings, cl compiler

### PR DESCRIPTION
**Pull request by ntfshard**
### Details:
 - Fix warnings spotted by cl compiler

### CODE REVIEW

File: inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.hpp [hunk 1]
```diff
@@ -13,6 +13,7 @@
 #include <ngraph/pass/pass.hpp>
 #include <ngraph/opsets/opset6.hpp>
 
+#include "details/ie_exception.hpp"
```
Comment (ilya-lavrenov) [inline inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.hpp:16]: can we remove this header usage at all from all sources? exception stuff is a part of ie_common.h how

File: ngraph/core/include/ngraph/op/constant.hpp [hunk 2]
```diff
@@ -646,24 +646,26 @@ namespace ngraph
                 }
                 template <
                     ngraph::element::Type_t Type,
+                    typename ValueT,
                     typename std::enable_if<Type == ngraph::element::Type_t::u4, bool>::type = true>
-                static ngraph::fundamental_type_for<Type>
-                    value_in_range(const ngraph::fundamental_type_for<Type>& value)
+                static ngraph::fundamental_type_for<Type> value_in_range(const ValueT& value)
```
Comment (ilyachur) [inline ngraph/core/include/ngraph/op/constant.hpp:651]: What warning did you solve here?

Comment (ntfshard) [reply ngraph/core/include/ngraph/op/constant.hpp:651]: Implicit type conversion of function argument.

**Comment by ntfshard** [comment]
BTW, build log size with and without this patch is 12.3MB and 1.4MB

**Review by ilyachur — APPROVED** [review]

**Pull request merged**